lib/susedistribution.pm

```diff
@@ -320,7 +320,7 @@ sub ensure_installed {
         last if (match_has_tag('pkcon-finished'));
         if (match_has_tag('Policykit')) {
             type_password;
-            send_key 'ret';
+            assert_and_click("PolicyKit-authenticate");
```

Comment:
That's a bit redundant as you introduce another needle which is basically the same as "PolicyKit", e.g. compare https://openqa.opensuse.org/tests/1268145#step/gnome_music/8 and https://openqa.opensuse.org/tests/1268145#step/gnome_music/9 . This is 1. more complicated than necessary, 2. inefficient for test execution, 3. double maintenance for needles. You can combine the two with one of the following approaches:

1. Just use the 'PolicyKit' needle and click on it when found with `click_lastmatch`
2. Just use the 'PolicyKit' needle and click on it using `assert_and_click` (though this triggers a double screen check but it is at least the same needle)
3. Same as 1. but use 'PolicyKit-authenticate' only

I prefer 3., e.g.:

```perl
if (match_has_tag('PolicyKit-authenticate')) {
    type_password;
    click_lastmatch;
}
```

The needle tag `PolicyKit-authenticate` should make it more obvious to needle creators that the "authenticate" button needs to be selected.

Reply:
Thanks - This is very valuable and I like it a lot. Changed the code to variant 3 (as you say, it DOES make it more obvious for a needle-creator)

Restarted a VR